pkg/zonemgr/zone_file.go

```diff
@@ -19,3 +17,7 @@ func NewZoneFile(fileName string) *ZoneFile {
 func (zoneFile *ZoneFile) writeFile(content string) (err error) {
 	return os.WriteFile(zoneFile.zoneFileFullName, []byte(content), zoneFilePerm)
 }
+
+func readFile(zoneFileName string) ([]byte, error) {
```

Comment:
Because it's a "util" function,  I use it inside ZoneFileCache in order to fetch SoaSerial. 
In the previous version this method was part of ZoneFile and ZoneManager did the job since it holds ZoneFile instance. 
But Or suggested to move this logic into ZoneFileCache file and use readFile as a helper function.
Here is the conversation:
https://github.com/kubevirt/kubesecondarydns/pull/45#discussion_r1043114185

I can create a util file for readFile function, I just wanted to spare creating another file just for one method that anyway matches ZoneFile functionality

Reply:
You can strive to make it as an API or so please (as Alona suggested https://github.com/kubevirt/kubesecondarydns/pull/45#pullrequestreview-1212967459)

Reply:
I prefer the previous design. I find it cleaner that the ZoneCache is not familiar with the ZoneFile.

Reply:
ZoneFile is a general util it is totally fine imo, 
there should not be two places that determine the serial and rebuild the header
the flow was much harder to grasp

Reply:
Why not reading the SOA from the file via ZoneManager and passing it to the ZoneCache?

Reply:
This is an option, this is not the previous design 
as long as we set initial serial once and build the header once without rebuild in case a serial exists, it is good imo

Reply:
The problem is with the zone file name. It is initiated based on the domain name. Which is set inside the ZoneCache. So in current design you can't read the SOA from the file and pass it to ZoneCache without first initializing the ZoneCache. Perhaps it can be changed by changing the design slightly, not sure it's worth it

Reply:
You can read the SOA from the file because you only look for `SOA.* ( ....` right ?
the regex doesn't depend on the ORIGIN / domain name
if so, the new helper function can be an API of ZoneFile and used by ZoneManager
unless i miss something

Reply:
You can find the only file in the folder, if exist by scanning the folder
if we all like the idea
then you don't need the name of it
if you want to bullet proof it, you can find the only file that starts with db in the folder etc